File: src/MergeAlternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        merged = []
        len1, len2 = len(word1), len(word2)
        min_len = min(len1, len2)

        for i in range(min_len):
            merged.append(word1[i])
            merged.append(word2[i])

        if len1 > len2:
            merged.append(word1[min_len:])
        else:
            merged.append(word2[min_len:])

        return ''.join(merged)
    

Solution().mergeAlternately("abc", "pqr")  #
# "apbqcr"
Solution().mergeAlternately("ab", "pqrs")  #
# "apbqrs"

File: practice/MergeAlternately.java
public class MergeAlternately {
    public String mergeAlternately(String word1, String word2) {
       int l1 = word1.length();
       int l2= word2.length();
        int d=Math.min(l1,l2);
        String s="";
        for(int i=0; i<d; i++){
            s = s + word1.charAt(i) + word2.charAt(i);
            if (i==d-1 && l1!=l2){
                if (l1>l2){
                    s=s+word1.substring(d);
                } else {
                    s=s+word2.substring(d);
                }
            }
        }
        return s;
    }
}